feat. Host SignUp/SignIn

HunCY5 opened the pull request:
- ProfileView에서 호스트 로그인 버튼 선택 시, HostLoginViewController로 이동
- 회원가입 시, type 필드 값 guest or host 로 저장되도록 추가
- 공공데이터 포털: 사업자등록정보 진위확인 및 상태조회 api 사용
- 현재 운영 중인 사업자 등록번호 일 경우에만 승인.
- 호스트 전용 MainTabBarController 구현
- 탭: 예약관리, 체크인, 객실관리 구성
- LoginModel: expectedType 파라미터로 사용자 타입을 검증
- 계정 타입에 따른 루트뷰 분기 로직 추가

HunCY5 committed:
feat. HostLoginView

- ProfileView에서 호스트 로그인 버튼 선택 시, HostLoginViewController로 이동
- LoginModel: type필드 기반으로 guest/host 로그인 분기 처리

HunCY5 committed:
feat. Guest SignUp

- type필드 값 guest로 저장되도록 추가

HunCY5 committed:
feat. LoginView Alert

- 로그인 뷰 상황별 알림 추가

HunCY5 committed:
feat. HostSignUpView UI

- 이메일, 사입자등록번호 검증 완료 시 필드+버튼 비활성화
- 회원가입 완료 시 type 필드 값: host

HunCY5 committed:
feat. 사업자등록번호 검증

- 공공데이터 포털: 사업자등록정보 진위확인 및 상태조회 api 사용
- 현재 운영 중인 사업자 등록번호 일 경우에만 승인.
- UIScrollView 적용

HunCY5 committed:
feat. Host SignUp

- Firebase Authentication 회원가입 로직 추가
- 각 필드별 입력 길이 제한

HunCY5 committed:
feat. HostMainTabBar

- 호스트 전용 MainTabBarController 구현
- 탭: 예약관리, 체크인, 객실관리 구성
- HostLoginView에서 로그인 진행 시, 루트 뷰 설정

HunCY5 committed:
feat. 걔정 타입 불일치 로그인 예외 처리

- LoginModel: expectedType 파라미터로 사용자 타입을 검증
- 게스트가 호스트로, 호스트가 게스트로 로그인 시 에러 메시지 표시

HunCY5 committed:
feat. SceneDelegate

- 계정 타입에 따른 루트뷰 분기 로직 추가

ChanSolShin commented:
에러목록
- 호스트 회원가입 이후 마이페이지로 돌아오면 게스트로 로그인이 됨
- 전화번호 예외처리 추가. 현재 영문으로 전화번호를 입력해도 가입가능
- 사업자 등록번호 키보드 열리면 뷰 깨짐
- 게스트/호스트 로그인페이지에서 필드입력후 키보드가 내려가지 않음

HunCY5 committed:
fix. HostSignUpViewController

- 호스트 회원가입 후 게스트 마이페이지로 이동 시, 로그인이 되는 오류 해결
- 회원가입 직후, 로그아웃 처리

HunCY5 committed:
fix. 키보드 회피 로직 개선

- 키보드 활성화 시, 필드가 위로 이동하면서 타이틀 텍스트 영역 침범 문제 해결
- keyboardWillShow 내에서 스크롤뷰 인셋과 필드 위치를 계산하여 스크롤되도록 수정

HunCY5 committed:
feat. Login Keyboard 완료 버튼

- 키보드에 완료 버튼을 추가하여 비활성화 할 수 있도록 추가

HunCY5 committed:
fix. 키보드 회피 로직 개선

사업자등로번호 필드가 활성화 되었을 때만 필드가 올라가도록 수정
